utils/model_uploader/model_autotracing.py

```diff
@@ -0,0 +1,452 @@
+# files for uploading to OpenSearch model hub.
+
+import argparse
+import os
+import shutil
+import sys
+import warnings
+from typing import List, Optional, Tuple
+from zipfile import ZipFile
+
+import numpy as np
+from numpy.typing import DTypeLike
+from sentence_transformers import SentenceTransformer
+
+THIS_DIR = os.path.dirname(__file__)
+ROOT_DIR = os.path.join(THIS_DIR, "../..")
+sys.path.append(ROOT_DIR)  # Required for importing OPENSEARCH_TEST_CLIENT
+
+LICENSE_PATH = "LICENSE"
+from opensearch_py_ml.ml_commons import MLCommonClient
+from opensearch_py_ml.ml_models.sentencetransformermodel import SentenceTransformerModel
+from tests import OPENSEARCH_TEST_CLIENT
+
+BOTH_FORMAT = "BOTH"
+TORCH_SCRIPT_FORMAT = "TORCH_SCRIPT"
+ONNX_FORMAT = "ONNX"
+
+ORIGINAL_FOLDER_PATH = "sentence-transformers-original/"
+TORCHSCRIPT_FOLDER_PATH = "sentence-transformers-torchscript/"
+ONNX_FOLDER_PATH = "sentence-transformers-onnx/"
+UPLOAD_FOLDER_PATH = "upload/"
+MODEL_CONFIG_FILE_NAME = "ml-commons_model_config.json"
+TEST_SENTENCES = [
+    "First test sentence",
+    "This is a very long sentence used for testing model embedding outputs.",
+]
+RTOL_TEST = 1e-03
+ATOL_TEST = 1e-05
+ML_BASE_URI = "/_plugins/_ml"
+
+
+def trace_sentence_transformer_model(
+    model_id: str,
+    model_version: str,
+    model_format: str,
+    embedding_dimension: Optional[int] = None,
+    pooling_mode: Optional[str] = None,
+) -> Tuple[str, str]:
+    """
+    Trace the pretrained sentence transformer model, create a model config file,
+    and return a path to the model file and a path to the model config file required for model registration
+
+    :param model_id: Model ID of the pretrained model
+    :type model_id: string
+    :param model_version: Version of the pretrained model for registration
+    :type model_version: string
+    :param model_format: Model format ("TORCH_SCRIPT" or "ONNX")
+    :type model_format: string
+    :param embedding_dimension: Embedding dimension input
+    :type embedding_dimension: int
+    :param pooling_mode: Pooling mode input ("CLS", "MEAN", "MAX", "MEAN_SQRT_LEN" or None)
+    :type pooling_mode: string
+    """
+    folder_path = (
+        TORCHSCRIPT_FOLDER_PATH
+        if model_format == TORCH_SCRIPT_FORMAT
+        else ONNX_FOLDER_PATH
+    )
+
+    # 1.) Initiate a sentence transformer model class object
+    pre_trained_model = None
+    try:
+        pre_trained_model = SentenceTransformerModel(
+            model_id=model_id, folder_path=folder_path, overwrite=True
+        )
+    except Exception as e:
+        assert (
+            False
+        ), f"Raised Exception in tracing {model_format} model\
+                             during initiating a sentence transformer model class object: {e}"
+
+    # 2.) Save the model in the specified format
+    model_path = None
+    try:
+        if model_format == TORCH_SCRIPT_FORMAT:
+            model_path = pre_trained_model.save_as_pt(
```

Comment:
How are we planning to add the LICENSE file?

Reply:
It's added in `prepare_files_for_uploading` step.